File: gunhoflash/climbing-stairs.ts
function climbStairs(n: number): number {
  let prevPrevStep = 0;
  let prevStep = 1;
  let step = 0;

  while (n--) {
    step = prevPrevStep + prevStep;
    prevPrevStep = prevStep;
    prevStep = step;
  }

  return step;
}

File: gunhoflash/climbing-stairs.py
class Solution:
  def climbStairs(self, n: int) -> int:
    prevPrevStep = 0
    prevStep = 1
    step = 0

    while n:
      n -= 1
      step = prevPrevStep + prevStep
      prevPrevStep = prevStep
      prevStep = step

    return step
